<!--
    ROS navigation stack with velocity smoother and safety (reactive) controller
-->
<launch>
 <arg name="tbName" default="asd123"/>

  <!-- <include file="$(find turtlebot_navigation)/launch/includes/velocity_smoother.launch.xml"/>
  <include file="$(find turtlebot_navigation)/launch/includes/safety_controller.launch.xml"/> -->

  <arg name="odom_frame_id"   default="$(arg tbName)/odom"/>
  <arg name="base_frame_id"   default="$(arg tbName)/base_footprint"/>
  <arg name="global_frame_id" default="map"/>
  <arg name="odom_topic" default="/$(arg tbName)/odom" />
  <arg name="laser_topic" default="/$(arg tbName)/scan" />
  <arg name="custom_param_file" default="$(find multi_robot)/param/dummy.yaml"/>

  <node pkg="move_base" type="move_base" respawn="false" name="move_base_$(arg tbName)" output="screen">
    <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" />
    <rosparam file="$(find turtlebot3_navigation)/param/costmap_common_params_burger.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find turtlebot3_navigation)/param/costmap_common_params_burger.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find turtlebot3_navigation)/param/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find turtlebot3_navigation)/param/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find turtlebot3_navigation)/param/move_base_params.yaml" command="load" />
    <rosparam file="$(find turtlebot3_navigation)/param/dwa_local_planner_params_burger.yaml" command="load" />
    <!-- external params file that could be loaded into the move_base namespace -->
    <rosparam file="$(arg custom_param_file)" command="load" />
    <param name="DWAPlannerROS/min_vel_x" value="0.0" if="false" />
    <!-- reset frame_id parameters using user input data -->
    <param name="global_costmap/global_frame" value="$(arg global_frame_id)"/>
    <param name="global_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
    <param name="local_costmap/global_frame" value="$(arg odom_frame_id)"/>
    <param name="local_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
    <param name="DWAPlannerROS/global_frame_id" value="$(arg odom_frame_id)"/>

    <remap from="cmd_vel" to="/$(arg tbName)/cmd_vel"/>
    <remap from="odom" to="$(arg odom_topic)"/>
    <remap from="scan" to="$(arg laser_topic)"/>
    <remap from="map" to="/map" />
    <remap from="/move_base_simple/goal"                                           to="/$(arg tbName)/move_base_simple/goal" />
    <remap from="/move_base/TebLocalPlannerROS/global_plan"                        to="/$(arg tbName)/move_base/TebLocalPlannerROS/global_plan" />
    <remap from="/move_base/TebLocalPlannerROS/local_plan"                         to="/$(arg tbName)/move_base/TebLocalPlannerROS/local_plan" />
    <remap from="/move_base/TebLocalPlannerROS/teb_markers"                        to="/$(arg tbName)/move_base/TebLocalPlannerROS/teb_markers" />
    <remap from="/move_base/TebLocalPlannerROS/teb_markers_array"                  to="/$(arg tbName)/move_base/TebLocalPlannerROS/teb_markers_array" />
    <remap from="/move_base/TebLocalPlannerROS/teb_poses"                          to="/$(arg tbName)/move_base/TebLocalPlannerROS/teb_poses" />
    <remap from="/move_base/global_costmap/costmap"                                to="/$(arg tbName)/move_base/global_costmap/costmap" />
    <remap from="/move_base/global_costmap/costmap_updates"                        to="/$(arg tbName)/move_base/global_costmap/costmap_updates" />
    <remap from="/move_base/local_costmap/costmap"                                 to="/$(arg tbName)/move_base/local_costmap/costmap" />
    <remap from="/move_base/local_costmap/costmap_updates"                         to="/$(arg tbName)/move_base/local_costmap/costmap_updates" />
    <remap from="/move_base/local_costmap/footprint"                               to="/$(arg tbName)/move_base/local_costmap/footprint" />

    <remap from="/move_base/GlobalPlanner/parameter_descriptions"                  to="/$(arg tbName)/move_base/GlobalPlanner/parameter_descriptions" />
    <remap from="/move_base/GlobalPlanner/parameter_updates"                       to="/$(arg tbName)/move_base/GlobalPlanner/parameter_updates" />
    <remap from="/move_base/GlobalPlanner/plan"                                    to="/$(arg tbName)/move_base/GlobalPlanner/plan" />
    <remap from="/move_base/GlobalPlanner/potential"                               to="/$(arg tbName)/move_base/GlobalPlanner/potential" />
    <remap from="/move_base/TebLocalPlannerROS/obstacles"                          to="/$(arg tbName)/move_base/TebLocalPlannerROS/obstacles" />
    <remap from="/move_base/TebLocalPlannerROS/parameter_descriptions"             to="/$(arg tbName)/move_base/TebLocalPlannerROS/parameter_descriptions" />
    <remap from="/move_base/TebLocalPlannerROS/parameter_updates"                  to="/$(arg tbName)/move_base/TebLocalPlannerROS/parameter_updates" />
    <remap from="/move_base/cancel"                                                to="/$(arg tbName)/move_base/cancel" />
    <remap from="/move_base/current_goal"                                          to="/$(arg tbName)/move_base/current_goal" />
    <remap from="/move_base/feedback"                                              to="/$(arg tbName)/move_base/feedback" />
    <remap from="/move_base/global_costmap/footprint"                              to="/$(arg tbName)/move_base/global_costmap/footprint" />
    <remap from="/move_base/global_costmap/inflation_layer/parameter_descriptions" to="/$(arg tbName)/move_base/global_costmap/inflation_layer/parameter_descriptions" />
    <remap from="/move_base/global_costmap/inflation_layer/parameter_updates"      to="/$(arg tbName)/move_base/global_costmap/inflation_layer/parameter_updates" />
    <remap from="/move_base/global_costmap/obstacle_layer/clearing_endpoints"      to="/$(arg tbName)/move_base/global_costmap/obstacle_layer/clearing_endpoints" />
    <remap from="/move_base/global_costmap/obstacle_layer/parameter_descriptions"  to="/$(arg tbName)/move_base/global_costmap/obstacle_layer/parameter_descriptions" />
    <remap from="/move_base/global_costmap/obstacle_layer/parameter_updates"       to="/$(arg tbName)/move_base/global_costmap/obstacle_layer/parameter_updates" />
    <remap from="/move_base/global_costmap/parameter_descriptions"                 to="/$(arg tbName)/move_base/global_costmap/parameter_descriptions" />
    <remap from="/move_base/global_costmap/parameter_updates"                      to="/$(arg tbName)/move_base/global_costmap/parameter_updates" />
    <remap from="/move_base/global_costmap/static_layer/parameter_descriptions"    to="/$(arg tbName)/move_base/global_costmap/static_layer/parameter_descriptions" />
    <remap from="/move_base/global_costmap/static_layer/parameter_updates"         to="/$(arg tbName)/move_base/global_costmap/static_layer/parameter_updates" />
    <remap from="/move_base/goal"                                                  to="/$(arg tbName)/move_base/goal" />
    <remap from="/move_base/local_costmap/obstacle_layer/parameter_descriptions"   to="/$(arg tbName)/move_base/local_costmap/obstacle_layer/parameter_descriptions" />
    <remap from="/move_base/local_costmap/obstacle_layer/parameter_updates"        to="/$(arg tbName)/move_base/local_costmap/obstacle_layer/parameter_updates" />
    <remap from="/move_base/local_costmap/parameter_descriptions"                  to="/$(arg tbName)/move_base/local_costmap/parameter_descriptions" />
    <remap from="/move_base/local_costmap/parameter_updates"                       to="/$(arg tbName)/move_base/local_costmap/parameter_updates" />
    <remap from="/move_base/local_costmap/static_layer/parameter_descriptions"     to="/$(arg tbName)/move_base/local_costmap/static_layer/parameter_descriptions" />
    <remap from="/move_base/local_costmap/static_layer/parameter_updates"          to="/$(arg tbName)/move_base/local_costmap/static_layer/parameter_updates" />
    <remap from="/move_base/parameter_descriptions"                                to="/$(arg tbName)/move_base/parameter_descriptions" />
    <remap from="/move_base/parameter_updates"                                     to="/$(arg tbName)/move_base/parameter_updates" />
    <remap from="/move_base/result"                                                to="/$(arg tbName)/move_base/result" />
    <remap from="/move_base/status"                                                to="/$(arg tbName)/move_base/status" />
    <remap from="/move_base_simple/goal"                                           to="/$(arg tbName)/move_base_simple/goal" />
  </node>


</launch>